De-duplicate some peek methods

dstansby opened the pull request:
Where timeseries source `.peek()` methods have a compatible signature with `GenericTimeSeries.peek()`, I've de-duplicated them. This also adds the `title` keyword argument to `GenericTimeSeries.peek()`.

nabobalis commented:
Looks like we got:

````
<unknown>:862: WARNING: py:meth reference target not found: sunpy.timeseries.sources.RHESSISummaryTimeSeries.peek
```

dstansby commented:
Thanks for spotting that - it was broken by this PR because I've removed `sunpy.timeseries.sources.RHESSISummaryTimeSeries.peek`. I've removed the doc link (it's in the changelog)
